examples/sparse/hgnn.py

```diff
@@ -0,0 +1,91 @@
+"""
+Hypergraph Neural Networks (https://arxiv.org/pdf/1809.09401.pdf)
+"""
+import dgl
+import dgl.data
+import dgl.mock_sparse as dglsp
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from torchmetrics.functional import accuracy
+import tqdm
+
+class HGNN(nn.Module):
+    def __init__(self, H, in_size, out_size, hidden_dims=16):
+        super().__init__()
+
+        self.Theta1 = nn.Linear(in_size, hidden_dims)
+        self.Theta2 = nn.Linear(hidden_dims, out_size)
+        self.dropout = nn.Dropout(0.5)
+
+        # Compute the Laplacian with Sparse Matrix API
+        d_V = H.sum(1)  # node degree
+        d_E = H.sum(0)  # edge degree
+        n_edges = d_E.shape[0]
+        D_V_invsqrt = dglsp.diag(d_V ** -0.5)  # D_V ** (-1/2)
+        D_E_inv = dglsp.diag(d_E ** -1)  # D_E ** (-1)
+        W = dglsp.identity((n_edges, n_edges))
+        self.laplacian = D_V_invsqrt @ H @ W @ D_E_inv @ H.T @ D_V_invsqrt
+
+    def forward(self, X):
+        X = self.laplacian @ self.Theta1(self.dropout(X))
+        X = F.relu(X)
+        X = self.laplacian @ self.Theta2(self.dropout(X))
+        return X
+
+def train(model, opt, X, Y, train_mask):
+    model.train()
+    Y_hat = model(X)
+    loss = F.cross_entropy(Y_hat[train_mask], Y[train_mask])
+    opt.zero_grad()
+    loss.backward()
+    opt.step()
+
+def evaluate(model, X, Y, val_mask, test_mask):
+    model.eval()
+    Y_hat = model(X)
+    val_acc = accuracy(Y_hat[val_mask], Y[val_mask])
+    test_acc = accuracy(Y_hat[test_mask], Y[test_mask])
+    return val_acc, test_acc
+
+def load_data():
+    dataset = dgl.data.CoraGraphDataset()
+
+    graph = dataset[0]
+    # The paper created a hypergraph from the original graph. For each node in
```

Comment:
Should this be on the top of the paper?
It sounds like a clarification for the whole implementation instead of a piece of code here.